var/spack/repos/builtin/packages/librelp/package.py

```diff
@@ -22,5 +22,6 @@ class Librelp(AutotoolsPackage):
     depends_on('automake', type='build')
     depends_on('libtool',  type='build')
     depends_on('m4',       type='build')
+    depends_on('valgrind', type='test')
```

Comment:
Is this test dependency picked up automatically by the `configure` script?

Reply:
No, but some test will fail without this software like that:
```
=========================================
   librelp 1.7.0: tests/test-suite.log
=========================================

# TOTAL: 28
# PASS:  25
# SKIP:  1
# XFAIL: 0
# FAIL:  2
# XPASS: 0
# ERROR: 0

.. contents:: :depth: 2

SKIP: duplicate-receiver-vg
===========================

This test currently does not do anything really useful. We should
either change that or drop the test.
I assume it shall test that librelp does not segfault if the
listen port cannot be bound to.
SKIP duplicate-receiver-vg.sh (exit status: 77)

FAIL: basic-sessionbreak-vg
===========================

============================================================
09:11:12: Test: ./basic-sessionbreak-vg.sh
============================================================
Starting Receiver...
got 102607 lrtb_42229.receive.pid
/home/spack-develop/opt/spack/linux-debian10-aarch64/gcc-8.3.0/libtool-2.4.6-2cthedktzpu4boy5c3bo2te2pxgfgeyh/bin/libtool: line 11632: exec: valgrind: not found
ABORT! Timeout waiting on startup, pid file lrtb_42229.receive.pid
FAIL basic-sessionbreak-vg.sh (exit status: 1)

FAIL: tls-basic-vg
==================

============================================================
09:11:53: Test: ./tls-basic-vg.sh
============================================================

BEGIN SUBTEST using TLS lib: gnutls
Starting Receiver...
got 103061 lrtb_44769.receive.pid
/home/spack-develop/opt/spack/linux-debian10-aarch64/gcc-8.3.0/libtool-2.4.6-2cthedktzpu4boy5c3bo2te2pxgfgeyh/bin/libtool: line 11632: exec: valgrind: not found
ABORT! Timeout waiting on startup, pid file lrtb_44769.receive.pid
FAIL tls-basic-vg.sh (exit status: 1)
```

And we resolve a similar issue in https://github.com/spack/spack/pull/21522

Reply:
Thanks, makes sense like you say if tests are run with shell scripts.